### ISSUE TITLE
Generate a cluster map based on dirents

### ISSUE THREAD

**Issue by burninrubber0** [aerosoul94/FATXTools#22]
Implement an option to generate a cluster map based on the location and size fields in dirents, while also un-deleting any deleted dirents. FATXTools already recovers files based on dirents when using the orphan analysis/metadata analyzer, so dirents should work well for this purpose.

Usually, when a formatted drive is received, the cluster map has been reset. This feature would effectively automate most of the busywork involved in the proper recovery of fragmented files, i.e. creating a new cluster map. Rather than having to set up an entire cluster chain manually to recover a file, only the fragments would need to be taken care of. This is likely a necessary prerequisite to #10 as the recovery of fragmented files would require the same busywork regardless of the existence of a cluster map editor.

**Comment by aerosoul94** [comment]
Cluster chain's are already generated for any **deleted** dirent's. Deleted dirent's are assigned a generated cluster chain while active (undeleted) files retain their original cluster chains from the FAT.
https://github.com/aerosoul94/FATXTools/blob/de05bd5b283ed4590f2f06599aea7f97e08a8a75/FATXTools/Database/FileDatabase.cs#L214-L221

Just for further explanation, we can see here how the cluster chains are generated by FATXTools. For directories, its a bit more dangerous to assume anything about its cluster chain, so we only assign a single cluster using the dirent's `FirstCluster` field. As for files, we generate a cluster chain of contiguous clusters starting from `FirstCluster` using `FileSize / ClusterSize` as the number of clusters.
https://github.com/aerosoul94/FATXTools/blob/de05bd5b283ed4590f2f06599aea7f97e08a8a75/FATXTools/Database/FileDatabase.cs#L157-L173

Btw, I also added a cluster chain editor although at current state, I'll admit it is a little unstable and definitely needs more testing. It's purpose is to allow a user to virtually modify the cluster chain without touching the FAT in the hdd image. But you can access it by right clicking a file and then clicking Edit Cluster Chain.

**Comment by burninrubber0** [comment]
That makes sense, not sure why I thought cluster chains weren't being generated. It's identical to what I've been doing manually, which is nice to see.

The idea behind opening this issue was to be able to save cluster map changes to the image in order to properly recover fragmented files, but that's not necessary anymore as the cluster chain can be generated/modified and have the changes saved without altering the image. (For reference, the last time I used FATXTools was when the only usable version was written in python, so I'm very out of the loop.)

That said, there is a potential use case for saving changes to the image, that being to restore a repaired version of an image to a hard drive. However, this is a very niche use case, and one I personally wouldn't use, so I understand if it doesn't get implemented. Up to you whether you want to pursue it; if not, this issue can be closed.

About the new cluster editor, I tested it and have a couple ideas for it, but will save those for a different thread. It works well enough at the moment, though.
